### PR TITLE
Handle relation-fields for working copies of folderish content

### LINKED ISSUE #118: Relations not handled for folderish content
When creating a working copy of folderish content the `RelationValues` are simply copied from the baseline to the copy. This reuses the same object for baseline and copy and since `__parent__` and `from_object` of the `RelavionValue` is set each time the source is changed and the only object you can change is the copy the relation from the baseline is essentially removed from the relation catalog. If the view uses the relation-catalog (e.g by using `plone.api.relation`) to display outgoing relations all objects that are being worked on look like it do not have any.

Instead the working copy needs to get freshly created `RelationValue` items.

### PR DESCRIPTION
**Pull request by pbauer**
Fixes #118

### CODE REVIEW

**Review by sneridagh — APPROVED** [review]
LGTM, but I'm a bit rusty on backend matters, asking @davisagli and @ericof too!

**Comment by pbauer** [comment]
@jenkins-plone-org please run jobs

File: plone/app/iterate/base.py [hunk 1]
```diff
@@ -114,43 +112,6 @@ class BaseContentCopier:
...
-            ref_func = reference_tool.getReferences
-        try:
-            # Plone 4.1 or later
-            brains = ref_func(item, objects=False)
-        except TypeError:
-            # Plone 4.0 or earlier.  Nothing to fix here
-            return
-        for brain in brains:
-            if brain.getObject() is None:
-                reference_tool.uncatalog_object(brain.getPath())
```
Comment (davisagli) [inline plone/app/iterate/base.py:149]: I see this is dead code within plone.app.iterate. Hopefully there aren't subclasses out in the wild using these methods...

**Review by davisagli — COMMENTED** [review]
Hmm, the intersection of 2 dark corners of Plone (working copies and relations). I don't think I should approve this without a test.

**Comment by pbauer** [comment]
@davisagli I added a test

**Comment by pbauer** [comment]
@jenkins-plone-org please run jobs

**Review by jensens — APPROVED** [review]

**Pull request merged**